### ISSUE TITLE
At minor intersection straight doesn't display lanes (left direction is not missing in turn lanes)

### ISSUE THREAD

**Issue by mdykierek** [osmandapp/OsmAnd#21543]
### Description

Please check why this intersection does not have lane assist (there is a lane for straight and right into Dąbrowskiego):
https://osmand.net/map/navigate/?start=50.281157,18.940706&end=50.283377,18.936066&profile=car#17/50.28226/18.93598

### Steps to reproduce

Simulate or navigate given route

### Actual result

No lane assist. There is a required lane change to go straight. The road changes from 1 into two lanes with right designated for right turn and left for straight ahead.

### Expected result

I expect through and right lanes displayed with straight in green.

### Your Environment (required)

```
OsmAnd Version: 4.9.7
Android/iOS version: Android 14
Device model: Motorola Edge 40 Pro
Crash-Logs: ?
```

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced by navigating to the provided [coordinates](https://osmand.net/map/navigate/?start=50.281157,18.940706&end=50.283377,18.936066&profile=car#17/50.28226/18.93598):  
- **Start:** 50.281157,18.940706  
- **End:** 50.283377,18.936066  
- Using the **Car profile** and starting a navigation simulation.  

**Issue Observed:**  
OsmAnd does not display lane assist at the intersection where the road splits into two lanes (one for going straight and one for turning right).  

**Expected Behavior:**  
The lane assist should show straight-ahead lanes in **green**, similar to how it appears in the second screenshot.  

OsmAnd~ 5.0.0#3919m, released: 2024-12-10

<table style="width:100%">
  </tr>
  <tr>
    <th>Actual result</th>
    <th>Expected result</th>
  </tr>
  <tr>
<th><img src="https://github.com/user-attachments/assets/ad2b386e-51bc-4686-89a0-63b2cafec484"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/06550538-9451-494f-8493-955132c18eb7"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by mdykierek** [comment]
Same situation here: https://osmand.net/map/navigate/?start=50.263316,18.902930&end=50.263817,18.897893&profile=car#17/50.26410/18.89870
Should be straight (green) and right (into supermarket parking and then Wincentego Pośpiecha).

**Comment by vshcherb** [comment]
@yuriiurshuliak did you enable to show all turn lanes inside widget settings?

**Comment by yuriiurshuliak** [comment]
Yes, I have enabled the "Show all turn lanes" option in the widget settings. Despite this, the app still does not display lane assist at the specified intersection.  

It appears to be a persistent issue even with the correct settings. 

OsmAnd~ 5.0.0#4025m, released: 2024-12-30

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/261fca4a-119b-4552-a56c-28a9ac4a1a0f"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/1c6ff97d-02af-4580-b60d-c72742684da2"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vshcherb** [comment]
This will be hard to implement because 
https://www.openstreetmap.org/way/1241462682 has ```C|TR``` but it has 3 possible combinations turn left, straight, turn right. Turn left is also possible, so there are 3 combinations to consider and only 2 options available.

So this line of code where it fails https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java#L1460 (https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java#L2300) and no turn is assigned and it's ignored.

Probably this part could be smarter implemented in future

**Comment by mdykierek** [comment]
I'll update the intersection on the osm. In fact left turn is forbidden there. Additionally U-turn in the oppsite directoin is forbidden too. Nonetheless, this should be fixed.

**Comment by mdykierek** [comment]
One more thing you probably need to check and consider. In Poland, explicit through lane does not allow for turns, especially U-turns, so that would not allow left on that intersection. I presume in most Europe this could be the same. Don't know about the rest of the world.

**Comment by ivanPyrohivskyi** [comment]
Tests [Route1](https://osmand.net/map/navigate/?start=50.281157,18.940706&end=50.283377,18.936066&profile=car) [Route2](https://osmand.net/map/navigate/?start=50.263316,18.902930&end=50.263817,18.897893&profile=car)
